src/Microsoft.DotNet.Darc/src/DarcLib/GitHubClient.cs

```diff
@@ -417,6 +418,95 @@ public async Task CreateOrUpdatePullRequestCommentAsync(string pullRequestUrl, s
             }
         }
 
+        public async Task CreateOrUpdatePullRequestMergeStatusInfoAsync(string pullRequestUrl, IReadOnlyList<MergePolicyEvaluationResult.SingleResult> evaluations)
+        {
+            (string owner, string repo, int id) = ParsePullRequestUri(pullRequestUrl);
+            // Get the sha of the latest commit for the current PR
+            string prSha = (await Client.PullRequest.Get(owner, repo, id))?.Head?.Sha;
+            if (prSha == null) 
+            {
+                throw new InvalidOperationException("We cannot find the sha of the pull request");
+            }
+
+            // Get a list of all the merge policies checks runs for the current PR
+            List <CheckRun> existingChecksRuns = (await Client.Check.Run.GetAllForReference(owner, repo, prSha)).CheckRuns.Where(e => e.ExternalId.StartsWith("maestro-policy-")).ToList(); 
+
+            var toBeAdded = evaluations.Where(e => existingChecksRuns.All(c => c.ExternalId != $"maestro-policy-{e.MergePolicyName}-{prSha}"));
```

Comment:
```suggestion
            var toBeAdded = evaluations.Where(e => existingChecksRuns.All(c => c.ExternalId != CheckRunId(e, prSha)"));
```

Reply:
Please create CheckRunId(or other named) and use it everywhere when $"maestro-policy-{e.MergePolicyName}-{prSha}" is currently used